### PR TITLE
Re-write (type-hinting, documentation)

### PR DESCRIPTION
**Pull request by rperson1508**
## Code-related

- *Bumped version to 2.0*.

#### Breaking changes
- Rename public function names to match API naming
#### Other changes
- Typehinted most classes and functions
- Re-organised files and functions and classes

## Documentation
- Added documentation for readthedocs

## Misc
- Added requirements.txt for dependencies
- Updated README.md (to improve style)
- Updated LICENSE
- Added .gitignore
- Shift example in README to a seperate folder

## Github
- Added PR and Issue templates

### CODE REVIEW

File: examples/example.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import asyncio
+
+from unbelipy import UnbeliClient
+
+UNB_API_TOKEN = "Token generated from https://unbelievaboat.com/applications/"
+# Warning:
+# It is advisable to keep your token safely
+# .env files are an option but it requires another dependency, so it's not always warranted.
```
Comment (chrisdewa) [inline examples/example.py:8]: Maybe just put.
```py
# Warning:
# It's advisable that you set your token as an enviromental variable. 
# Always keep your credentials as a secret
 ```

File: unbelipy/__init__.py [hunk 2]
```diff
@@ -22,13 +22,16 @@
...
+
 import logging
 
-from .client import UnbeliClient
+from .client import UnbeliClient as UnbeliClient
 from .errors import *
 from .objects import (
-    UserBalance,
-    Guild
+    UserBalance as UserBalance,
```
[the remark this answers is not in the document]
Comment (rperson1508) [reply unbelipy/__init__.py:32]: So that typecheckers know it's an export

[the remark this answers is not in the document]
Comment (chrisdewa) [reply unbelipy/__init__.py:32]: Alright then

**Comment by rperson1508** [comment]
> I think it's better to remove all `__pycache__` files and folders, i added a template .gitignore that should help with that too.

Yeah forgot to leave those out
Will be off from my laptop for the next few days so will only make changes then

**Pull request merged**